### ISSUE TITLE
ServiceBus not receiving high number of messages within batches, and struggle to complete messages - no errors

### ISSUE THREAD

**Issue by andrewvi7** [Azure/azure-sdk-for-python#34338]
product:azsdk-python-servicebus, version:7.11.4],
framework:Python/3.10.12
platform:Linux-5.15.0-1054-azure-x86_64-with-glibc2.35
user-agent:azsdk-python-servicebus/7.11.4 uamqp/1.6.8 Python/3.10.12 (Linux-5.15.0-1054-azure-x86_64-with-glibc2.35)

Since the 22nd of January without code changes, the servicebus behaviour changed in a few ways:
1. When requesting batches of messages of 500, it used to bring back more or less 500 (when there were enough messages), now it fluctuates heavily, but averages around 80 per call. but goes down to as low as 8 messages.
2. Out of all the messages received, sometimes the number of completed messages fluctuates between 100%, to as low as 1% (saw a run that retrieved 500 messages and completed 6), but usually sits at around 50% without errors or any other explanation. Code executes within 2-4 seconds without message locks or other exceptions.

![image](https://github.com/Azure/azure-sdk-for-python/assets/59656660/cdd75c1f-6fd7-49b5-9ee7-7838dc04a93e)

![image](https://github.com/Azure/azure-sdk-for-python/assets/59656660/afdf7a13-e819-4796-b914-d45fe43fb41a)

Tried using the legacy protocol (uamqp_transport=True) which helps slightly, but still under performing to what it used to be like pre 22nd of Jan, although better than using the latest protocol, to where total received messages drops slightly, but the completed message rate drops even more.

Unsure on how to reproduce the issue as within the development servicebus under standard tier, the performance is much better and issue less noticable with the legacy protocol. However in production where there is a constant flow of messages, the issue is apparent, despite the servicebus being under premium tier, memory usually fluctuating around 30-60% and CPU highest usage sitting at under 10%.

**Expected behavior**
Despite the documentation stating that even when asking the service bus for max messages at 500, that the service bus will prioritise speed over the quota, it should still not be returning only 8 when there is 100k messages in the service bus. Pre 22nd of Jan, it pretty much always returned 500.

Also if x number of messages are read, the same amount should be completed or exception thrown for whatever reason. While as mentioned the code runs in 2-4 seconds, without any exceptions, but within the metrics I can clearly see it averaging 50% completion rate, and messages having delivery count incremented when peeking at messages through the portal SB explorer.

**Additional context**
Reached out to the support team, who mentioned that they were unable to find any errors on the service bus side, but also couldn't find any related issues on the function app that runs the pull down of messages.

Also tried pulling some uamqp logs out but can't see anything obvious, but wanted to also enquire about what these Deposition messages mean exactly as they seem to be generated from the app trying to complete the messages, of where there's far fewer of these deposition messages, than the "Transfer" ones:

![image](https://github.com/Azure/azure-sdk-for-python/assets/59656660/aeee28e3-dcda-4aeb-b48b-889f20c046d2)

![image](https://github.com/Azure/azure-sdk-for-python/assets/59656660/65152fbc-656e-491f-9e3e-6a3f30ca5bde)

**Comment by pvaneck** [comment]
Hey, thanks for the feedback, We'll investigate as soon as possible.

**Comment by kashifkhan** [comment]
Thank you for the detailed issue @andrewvi7. Has anything changed on your infrastructure side such as perhaps a different failover? Have the message sizes changed at all ?

**Comment by andrewvi7** [comment]
> Thank you for the detailed issue @andrewvi7. Has anything changed on your infrastructure side such as perhaps a different failover? Have the message sizes changed at all ?
> 
> In this particular case I would have to direct you towards opening a support ticket with the service team as the "it was working till Jan 22" part is quite telling. We havent updated the SDK since Nov 2023, so a sudden change is surprising and it would be best to start from there.

So most of the message sizes are around 20-80KB, but some go up to around 300-400KB, still no where near the limit of the premium tier.

Everything is pretty much the same, including code and infrastructure from what i can tell and the code is pretty basic to what is provided in the documentation samples, with some additional code to concat the messages, gzip them, write them into ADLS and go back over all the message to complete them.

We have reached out and raised a support ticket, but they've looked at the SB logs, and function app that runs the code and couldn't find anything that would explain the behaviour. So kind of reached a dead end in that regard.

Also forgot to include that i was seeing some strange behaviour changes like the below screenshot. I've excluded the count total messages in the topic as the graph becomes unreadable, but there was around 200k messages there when outgoing messages spike dropped around 5am.

![image](https://github.com/Azure/azure-sdk-for-python/assets/59656660/94576a02-7694-42a8-9cb8-f891b9beca2c)

Long term view:
![image](https://github.com/Azure/azure-sdk-for-python/assets/59656660/cc467c1b-4318-4032-baee-c38c460535b7)

The high spikes are essentially me running multiple iterations of the function app code in databricks to pull everything down.

And the last few days i've put in some code to do 5 calls for 500 messages in a single call to cope with the load so our production pipelines aren't severely impacted, but the issue is still there

I don't suppose there is a way to check the reason behind why a message wasn't completed and had it's delivery count incremented?
